autodist/kernel/synchronization/ps_synchronizer.py

```diff
@@ -208,7 +216,7 @@ def ctrl_consumers(op):
             raise RuntimeError("Incorrect old_grad.")
         return agg_grad
 
-    def _prune_control_dependencies(self, graph_item, var_op_name, master_replica=0):
+    def _prune_control_dependencies(self, graph_item, var_op_name, master_replica=0, optimize=False):
```

Comment:
overall I feel this PR changes  the signature of too many kernel methods. If there any way to keep the introduced `optimize` argument only in graph_item (cuz it is a status tracker of the graph_item object); other methods do not need to know its existence?